src/generator_http.rs

```diff
@@ -506,3 +512,69 @@ pub fn dispatch_pages(
         println!("dispatch_pages: stream already closed, all good");
     }
 }
+
+pub fn orchestrate(client: &DatagoClient) -> DatagoEngine {
```

Comment:
this is a big change @photoroman, each path owns the spawning of the subthreads and what the data pipes should look like, client code is smaller and just calls for "orchestrate". Was required with webdataset since the data pipe types change (tarball and binary payload), but I think that's also just a more flexible and cleaner design

Reply:
moved to other PR